persistence/src/main/scala/akka/persistence/EventStoreinterop.scala

```diff
@@ -0,0 +1,196 @@
+      extends Exception(s"Events buffer with capacity $bufferCapacity was overflowed, recovery for $persistenceId failed")
+      with NoStackTrace
+
+  /** Create instance of [[EventStore]] that uses Akka Persistence journal plugin under the hood.
+    *
+    * Journal plugin uses "push" model to recover events (ie read events from underline DB) while [[EventStore]] provides "pull" API via
+    * [[sstream.Stream]]. To overcome this limitation, the interop uses internal buffer to hold events provided by Akka' journal plugin
+    * before they will be consumed (ie deleted from buffer) as [[EventStore.events]] stream. The output stream is lazy by itself and actual
+    * event consumption from the buffer will happened only on the stream materialization.
+    *
+    * @param system
+    *   Akka system
+    * @param timeout
+    *   maximum time between messages from Akka' journal plugin (is the next message expected)
+    * @param capacity
+    *   internal event buffer capacity, on oveflow will raise [[BufferOverflowException]]
+    * @param journalPluginId
+    *   Akka persistence journal plugin ID
+    * @param eventSourcedId
+    *   Akka persistence ID, unique per each actor
+    * @return
+    *   instance of [[EventStore]]
+    */
+  def apply[F[_]: Async: FromFuture: ToTry, A](
+    system: ActorSystem,
+    timeout: FiniteDuration,
+    capacity: Int,
+    journalPluginId: String,
+    eventSourcedId: EventSourcedId
+  ): F[EventStore[F, A]] =
+    Sync[F]
+      .delay {
+        val actorRef = Persistence(system).journalFor(journalPluginId)
+        ActorEffect.fromActor(actorRef)
+      }
+      .map { journaller =>
+        new EventStore[F, A] {
+
+          import sstream.FoldWhile._
+
+          val persistenceId = eventSourcedId.value
+
+          override def events(fromSeqNr: SeqNr): F[sstream.Stream[F, EventStore.Persisted[A]]] = {
+
+            type Buffer = Vector[EventStore.Event[A]]
+
+            def actor(buffer: Ref[F, Buffer]) =
+              LocalActorRef[F, Unit, SeqNr]({}, timeout) {
+
+                case (_, JournalProtocol.ReplayedMessage(persisted)) =>
+                  if (persisted.deleted) ().asLeft[SeqNr].pure[F]
+                  else {
+                    val payload = persisted.payload.asInstanceOf[A]
+                    val event   = EventStore.Event(payload, persisted.sequenceNr)
+
+                    buffer
+                      .modify { buffer =>
+                        val buffer1 = buffer :+ event
+                        val lenght  = buffer.length + 1
+                        buffer1 -> lenght
+                      }
+                      .flatMap { lenght =>
+                        if (lenght > capacity) {
+                          new BufferOverflowException(capacity, persistenceId).raiseError[F, Either[Unit, SeqNr]]
+                        } else {
+                          ().asLeft[SeqNr].pure[F]
+                        }
+                      }
```

Comment:
This could also be written as following. I do not insist you should do it, though. Your way is totally fine too.
```scala
.ensure(new BufferOverflowException(capacity, persistenceId))(_ <= capacity)
.as(().asLeft[SeqNr])
```

Reply:
https://github.com/evolution-gaming/akka-effect/pull/276#discussion_r1450514138

Reply:
@dfakhritdinov It totally is fine to keep it your way, but, please, `lenght` -> `length` :blush:

Reply:
fixed, sorry for misunderstanding